frontend/src/Components/Story/StoryViewer.tsx

```diff
@@ -0,0 +1,92 @@
+import { Button, Collapse, Divider, Intent, Pre } from "@blueprintjs/core";
+import { useState } from "react";
+import {
+  StoryDeathLog,
+  StoryFoodLog,
+  StoryLog,
+  StoryMessageLog,
+  StoryPlatformLog,
+} from "../../Helpers/Logging/StoryLog";
+import { Ticker } from "./Ticker";
+
+interface StoryViewerProps {
+  story: StoryLog[];
+}
+
+export default function StoryViewer(props: StoryViewerProps) {
+  const { story } = props;
+  const [isOpen, setIsOpen] = useState(false);
+  return (
+    <div style={{ margin: "10px 0px" }}>
+      <Button
+        intent={isOpen ? Intent.PRIMARY : Intent.WARNING}
+        onClick={() => setIsOpen(!isOpen)}
+        style={{ width: 200 }}
+      >
+        {isOpen ? "Hide" : "Show"} Story
+      </Button>
+      <Collapse isOpen={isOpen} keepChildrenMounted={true}>
+        <Pre>
+          <StoryController story={story} />
+        </Pre>
+      </Collapse>
+    </div>
+  );
+}
+
+interface StoryControllerProps {
+  story: StoryLog[];
+}
+
+function StoryController(props: StoryControllerProps) {
+  const { story } = props;
+  const [tick, setTick] = useState(1);
+  const maxTick = Math.max(...story.map((e) => e.tick));
+
+  return (
+    <div>
+      <Ticker tick={tick} setTick={setTick} maxTick={maxTick} />
+      <Divider />
+      <div>
+        {story.map((log, index) => {
+          if (log.tick === tick) {
+            switch (log.msg) {
+              case "food":
+                const f = log as StoryFoodLog;
+                return (
+                  <div key={index}>
+                    Agent {f.atype} took {f.foodTaken} food on floor {f.floor} and left {f.foodLeft}
+                  </div>
+                );
+              case "message":
+                const m = log as StoryMessageLog;
+                return (
+                  <div key={index}>
+                    Agent {m.atype} on floor {m.floor} sent a message {m.mtype} targeting floor {m.target} with{" "}
```

Comment:
```suggestion
                    Agent {m.atype} on Floor {m.floor} sent a message {m.mtype} targeting Floor {m.target} with{" "}
```

Reply:
Dunno if it would be more easy to read if it was formatted as a table or with \t? Not a big deal if it's annoying to change though.

Reply:
I was gonna represent these in a UI. Can definitely represent in a more tabular format to make it easier to read or add an icon for specific actions at the start of the line. Will do something for it